clients/wallet/src/error.rs

```diff
@@ -31,6 +31,47 @@ pub enum Error {
 }
 
 impl Error {
+	pub fn is_recoverable(&self) -> bool {
+		match self {
+			Error::HttpFetchingError(e) if e.is_timeout() => true,
+			Error::HorizonSubmissionError { title: _, status, reason: _, envelope_xdr: _ }
+				if *status == 504 =>
+				true,
+			Error::CacheError(e) => match e.kind {
+				CacheErrorKind::CreateDirectoryFailed |
+				CacheErrorKind::FileCreationFailed |
+				CacheErrorKind::WriteToFileFailed |
+				CacheErrorKind::DeleteFileFailed |
+				CacheErrorKind::FileDoesNotExist => true,
+				_ => false,
+			},
+			_ => false,
+		}
+	}
+
+	pub fn is_server_error(&self) -> bool {
+		// error  500
+		let internal_server_error = reqwest::StatusCode::INTERNAL_SERVER_ERROR.as_u16();
+		// error 511
+		let net_auth_required = reqwest::StatusCode::NETWORK_AUTHENTICATION_REQUIRED.as_u16();
+
+		let check_status =
+			|status: u16| status >= internal_server_error && status <= net_auth_required;
+
+		match self {
+			Error::HttpFetchingError(e) => {
+				if let Some(status) = e.status() {
+					return check_status(status.as_u16())
+				}
+				false
+			},
```

Comment:
Maybe we can simplify this as follows? I guess you wanted to limit it to the status codes that actually exist but I guess we don't even need to do this limitation. As the server error responses range from [500-599](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses), we could just increase the range I suppose. WDYT @b-yap?
```suggestion
		match self {
			Error::HttpFetchingError(e) => {
				if let Some(status) = e.status() {
					let server_errors = 500..599;
					return server_errors.contains(&status)
				}
				false
			},
```

Reply:
done